core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -162,6 +161,25 @@ public ClusterState execute(ClusterState currentState) throws Exception {
         });
     }
 
+    void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, PutPipelineRequest request) throws Exception {
```

Comment:
This check is only for processors provided by plugin. If a pipeline has processors that aren't available out of the box then we need to check if all nodes in the cluster have these plugins. Otherwise a plugin can't be loaded on all nodes.